### PR TITLE
SEQNG-1126 Added exposure stage status to progress event.

### PR DESCRIPTION
**Pull request by jluhrs**
The current observation stage (preparing, acquiring, reading out) is now read from the instrument and sent to the web client.

### CODE REVIEW

File: modules/seqexec/server/src/main/resources/Gmos.xml [hunk 1]
```diff
@@ -469,10 +469,20 @@
...
+        <attribute name="obsPrep">
+            <channel>sad:dc:prep.VAL</channel>
+            <type>INTEGER</type>
+            <description>Observation is preparing</description>
+        </attribute>
         <attribute name="obsAcq">
             <channel>sad:dc:acq.VAL</channel>
             <type>INTEGER</type>
-            <description>Observation is aquiring</description>
+            <description>Observation is acquiring</description>
```
Comment (sraaphorst) [inline modules/seqexec/server/src/main/resources/Gmos.xml]: Why is it "Acquiring observation" for F2 but "Observation is acquiring" for Gmos? Should these be consistent?

Comment (sraaphorst) [reply modules/seqexec/server/src/main/resources/Gmos.xml]: (Same with others: Reading out observation vs. Observation is reading out and Preparing observation vs. Observation is preparing?)

Comment (jluhrs) [reply modules/seqexec/server/src/main/resources/Gmos.xml]: That's boring :)

File: modules/seqexec/server/src/main/scala/seqexec/server/ProgressUtil.scala [hunk 2]
```diff
@@ -30,12 +33,25 @@ object ProgressUtil {
       .collect{ case Some(p) => p }
 
   def countdown[F[_]: Applicative: Timer](total: Time, elapsed: Time): Stream[F, Progress] =
+    Stream.emit(ObsProgress(total, RemainingTime(total), ObserveStage.Preparing)).covary[F] ++
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/ProgressUtil.scala]: I think you can drop `covary` here

File: modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala [hunk 3]
```diff
@@ -267,6 +267,7 @@ trait ModelBooPicklers extends GemModelBooPicklers {
   implicit val userNotificationPickler    = generatePickler[UserNotification]
   implicit val guideConfigPickler         = generatePickler[GuideConfigUpdate]
   implicit val queueUpdatedPickler        = generatePickler[QueueUpdated]
+  implicit val observationStagePickler    = enumeratedPickler[ObserveStage]
```
Comment (cquiroz) [inline modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: Did you add a test for this?

**Comment by cquiroz** [comment]
What events will be sent to giapi based instruments?

File: modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala (hunk 3 above)
Comment (jluhrs) [reply modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/ModelBooPicklers.scala]: Yes.

File: modules/seqexec/server/src/main/scala/seqexec/server/ProgressUtil.scala (hunk 2 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/ProgressUtil.scala]: Done

**Comment by jluhrs** [comment]
@cquiroz There are no events "sent" to the instrument here.
GPI uses a simulated countdown. That countdown now goes through the stages Preparing-Acquiring-ReadingOut, so the client will not see a difference.

**Pull request merged**